Enable mingw CI job when compilation succeed

Currently, the mingw CI job fails with:
```
-- The C compiler identification is unknown
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Check for working C compiler: C:/msys64/mingw64/bin/cc.exe
CMake Error at C:/Program Files/CMake/share/cmake-3.31/Modules/CMakeTestCCompiler.cmake:67 (message):
  The C compiler
    "C:/msys64/mingw64/bin/cc.exe"
  is not able to compile a simple test program.
```
while attempting to generate the project solution. I investigated the issue but was unable to find a workable solution.
In order to not prevent other PR's from making progress (or rather, not confuse actual CI failures with this one), I went ahead and disabled the mingw CI job for the time being.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -406,8 +406,7 @@ jobs:
     mingw:
         # windows is 2x expensive to run on GitHub machines, so only run if we know something else simple passed as well
         needs: linux-no-asm
-        if: false # Disabled due to issues with msys2 making CMake unable to find a working compiler
-        runs-on: windows-2022
+        runs-on: windows-latest
         timeout-minutes: 30
         defaults:
             run:

